### PR TITLE
default vc to block v3 endpoint and deprecate block-v3 flag

### LINKED ISSUE #5040: Switch to block v3 by default in VC
## Description

Once sufficient client releases are out supporting block v3, or Deneb is safely activated on most networks, we should enable block v3 by default in the VC (see #4813 for discussion).

### PR DESCRIPTION
**Pull request by eserilev**
## Issue Addressed

Closes #5040

## Proposed Changes

Deprecate the produce-block-v3 flag, and use the v3 endpoint by default within the VC

This should only be merged after deneb is safely activated.

### CODE REVIEW

**Review by dapplion — APPROVED** [review]
LGTM

**Comment by michaelsproul** [comment]
I think we should wait until after v5.1.0 for this one

**Comment by michaelsproul** [comment]
Looks like we'll be OK to merge this for v5.2.0. There was a bug in Nimbus's v3 implementation prior to v24.4.0, which is due imminently:

- https://github.com/status-im/nimbus-eth2/commit/f20a21c01555b62aa747508cc6bb006a619c6998

Bug:

- https://github.com/status-im/nimbus-eth2/pull/6204

We should call this out in the release notes so that users running Nimbus BNs know they need to upgrade.

I'm doing some more testing with other BNs and blockdreamer.

**Comment by michaelsproul** [comment]
Idk about this. My Teku testing got blocked by a bug on their side involving eleel which I haven't had time to fix.

Let's just drop it from v5.2.0 for now so we can get moving.

**Review by michaelsproul — APPROVED** [review]
LGTM. Nice simplification.

**Comment by michaelsproul** [comment]
@mergify queue

**Comment by michaelsproul** [comment]
I'm not sure why the fallback simulator is failing. Haven't had a chance to investigate

**Comment by eserilev** [comment]
fwiw both the fallback and basic sim seem to work fine locally

**Comment by jimmygchen** [comment]
It looks like the BNs had some issues with getting payload from the EL. I've re-run the job to check if it's just a flaky error:

```
2024-07-22T19:21:17.0428622Z Jul 22 19:21:17.017 ERRO Error whilst producing block            info: block v3 proposal failed, this error may or may not result in a missed block, block_slot: Slot(110), error: "Error from beacon node when producing block: Recoverable(\"Error from beacon node when producing block: ServerMessage(ErrorMessage { code: 400, message: \\\"BAD_REQUEST: failed to fetch a block: GetPayloadFailed(EngineError(Api { error: PayloadIdUnavailable }))\\\", stacktraces: [] })\")", service: block, service: validator_2
```

File: validator_client/src/block_service.rs [hunk 1]
```diff
@@ -584,7 +511,7 @@ impl<T: SlotClock + 'static, E: EthSpec> BlockService<T, E> {
                         &metrics::BLOCK_SERVICE_TIMES,
                         &[metrics::BEACON_BLOCK_HTTP_GET],
                     );
-                    let block_response = Self::get_validator_block_v3(
+                    let block_response = Self::get_validator_block(
```
Comment (jimmygchen) [inline validator_client/src/block_service.rs]: Not part of this PR, I think it may be unnecessary to re-map the `BlockError` here?

https://github.com/sigp/lighthouse/pull/5292/files#diff-9397634273d7e6a89631ace49f8f1cfe7f083db2b556f1dee86b76184b7618e6R524-R529

**Comment by jimmygchen** [comment]
Oh I think the fallback is not working because we're always returning an Ok here instead of returning the result:

https://github.com/sigp/lighthouse/blob/02011199c3b42f36ce83f78aa8c82e20cc97a232/validator_client/src/block_service.rs#L531

File: testing/simulator/src/cli.rs [hunk 2]
```diff
@@ -77,7 +77,6 @@ pub fn cli_app() -> Command {
                 )
                 .arg(
                     Arg::new("vc-count")
-                        .short('c')
```
Comment (eserilev) [inline testing/simulator/src/cli.rs:80]: it seems we have two flags with the same short values
- continue-after-checks
- vc-count

this causes a build error so i've removed one of them

**Comment by eserilev** [comment]
thanks @jimmygchen! That seems to have fixed it (locally at least). The job failed in CI but maybe its just flaky? ill retry later when I'm at my computer

File: testing/simulator/src/cli.rs (hunk 2 above)
Comment (jimmygchen) [reply testing/simulator/src/cli.rs:80]: Yep looks fine to me - I don't see any usage of the short args

**Review by jimmygchen — APPROVED** [review]
Nice, green CI now 🎉

**Comment by jimmygchen** [comment]
@mergify queue
